File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/EvaluatorNewMemberAttendance.java
package com.ibcs.idsdp.rpm.model.domain;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.ibcs.idsdp.common.model.domain.BaseEntity;
import com.sun.istack.NotNull;

import lombok.Data;


@Data
@Entity
@Table(name = "m2_evaluator_new_member_attendance")
public class EvaluatorNewMemberAttendance extends BaseEntity{
	
	@NotNull
	@ManyToOne
	@JoinColumn(name = "m2_researcher_presentation_id")
	private ResearcherPresentation researcherPresentation;
	
	@NotNull
	@ManyToOne
	@JoinColumn(name = "m1_researcher_proposal_id")
	private ResearcherProposal researcherProposal;
	
	private Long stProfileOfExpertEvaluatorsId;

	@ManyToOne
	@JoinColumn(name = "m2_add_new_member_id")
	private NewMember newMember;	
	
	private Boolean isPresent;
	

}

File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/EvaluatorNewMemberAttendance.ts
import {ResearcherProposal} from "./ResearcherProposal";
import {ResearcherPresentation} from "./ResearcherPresentation";
import {NewMember} from "./NewMember";
import {ExpertEvaluator} from "../../settings/models/expert-evaluator.model";

export class EvaluatorNewMemberAttendance {

    id: number;
    uuid: string;
    m2ResearcherPresentationId: number;
    m1ResearcherProposalId: number;
    stProfileOfExpertEvaluatorsId: number;
    m2AddNewMemberId: number;
    isPresent: boolean;
    researcherPresentationResponseDto: ResearcherPresentation;
    researcherProposalResponseDto: ResearcherProposal;
    newMemberResponseDto: NewMember;
    expertEvaluatorResponseDto: ExpertEvaluator;
    researcherPresentation:  ResearcherPresentation;
    researcherProposal: ResearcherProposal;
}
